plugins/discovery-gce/src/main/java/org/elasticsearch/discovery/gce/GceHttpRequestInitializer.java

```diff
@@ -90,16 +90,23 @@ public void initialize(HttpRequest httpRequest) {
             httpRequest.setUnsuccessfulResponseHandler(credential);
         }
 
-        final long noTimeout = TimeValue.MINUS_ONE.getMillis();
-
-        final long connectTimeoutMillis = connectTimeout.getMillis();
-        if (connectTimeoutMillis > noTimeout) {
-            httpRequest.setConnectTimeout((int) connectTimeoutMillis);
+        final long infiniteTimeout = TimeValue.MINUS_ONE.getMillis();
+        if (connectTimeout != null) {
+            final long connectTimeoutMillis = connectTimeout.getMillis();
+            if (connectTimeoutMillis > 0) {
+                httpRequest.setConnectTimeout((int) connectTimeoutMillis);
+            } else if (connectTimeoutMillis == infiniteTimeout) {
```

Comment:
this smells like it should be a setting validation thing. Testing for this so deep, on every request without throwing exceptions feels wrong.

Reply:
Raaah yes you're right, I should have seen that. Thanks.